clean.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+usage()
+{
+    echo "Usage: clean [options]"
+    echo "Cleans the local dev environment."
+    echo
+    echo "  -b     Delete bin directory"
+    echo "  -p     Delete packages directory and NuGet package caches"
+    echo "  -a     Remove all untracked files in the working tree and delete NuGet package caches"
+    exit 1
+}
+
+# Parse arguments
+
+if [ $# == 0 ]
+then
+    echo "Please specify at least one option."
+    echo "Use 'clean -h' for help."
+    exit 1
+fi
+
+while [[ $# > 0 ]]
+do
+    opt="$1"
+    case $opt in
+        -h|--help)
+        usage
+        ;;
+        -b)
+        CleanBinDir=true
+        ;;
+        -p)
+        CleanPackagesDir=true
+        CleanPackagesCache=true
+        ;;
+        -a)
+        CleanWorkingTree=true
+        CleanPackagesCache=true
+        ;;
+        *)
+        echo "Unrecognized argument '$opt'"
+        echo "Use 'clean -h' for help."
+        exit 1
+        ;;
+    esac
+    shift
+done
+
+WorkingTreeRoot="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+if [ "$CleanBinDir" == true ] && [ "$CleanWorkingTree" != true ]
+then
+    echo "Removing bin directory"
+    rm -rf $WorkingTreeRoot/bin
+fi
+
+if [ "$CleanPackagesDir" == true ] && [ "$CleanWorkingTree" != true ]
+then
+    echo "Removing packages directory"
+    rm -rf $WorkingTreeRoot/packages
```

Comment:
It does seem like a bit of this is duplicated MSBuild logic. Have we considered adding the logic there, and then you can just use build.sh/cmd /t:Clean? Otherwise we'll have two sets of cleaning logic, more or less duplicated.